ha/fault_tolerance/event_manager.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+
+'''Module whoch provides interfaces to subscribe or unsubscribe for HA
+   events to cortx components'''
+
+import json
+
+from cortx.utils.log import Log
+from ha.core.config.config_manager import ConfigManager
+
+
+class EventManager:
+    '''Module responsible for publishing the HA event to a cortx component
+       who has subscribed for the specific HA event through message bus'''
+
+    def __init__(self):
+        '''Make initialization work for Event Manager'''
+        self._confstore = ConfigManager.get_confstore()
+        self.cortx_comp_list = ['sspl', 'csm', 's3', 'motr', 'hare', 'ha']
+        self._event_key = 'events/subscribe'
+
+    def subscribe(self, component : str, events : list):
+        '''
+           Register all the events for the notification. It maintains
+           list of events registered by the components. It maps the event
+           name with component name and store it in the consul
+           Ex:
+           key: /cortx/ha/v2/events/subscribe/sspl
+           value: ['enclosure:sensor:voltage', ...]
+        '''
+        Log.info(f'Received a subscribe request from {component} for event: {events}')
+        self._validate_component(component)
+        self._validate_events(events)
+        event_string = json.dumps(events)
+        if self._confstore.key_exists(f'self._event_key/{component}'):
+            Log.error(f'{component} has already subscribed')
+            raise Exception(f'{component} has already subescribed')
+        self._confstore.set(f'{self._event_key}/{component}', event_string)
```

Comment:
It might be a good idea to add the component in lower case to confstore.
So everywhere, while accessing, uniformity can be maintained. @indrajitzagade any thoughts ?

Reply:
Ok. Will check

Reply:
I think this is already handled